neqo-http3/src/connection_client.rs

```diff
@@ -5562,4 +5656,311 @@ mod tests {
             assert_closed(&client, &Error::HttpSettings);
         }
     }
+
+    #[test]
+    fn response_w_1xx() {
+        let (mut client, mut server, request_stream_id) = connect_and_send_request(true);
+
+        setup_server_side_encoder(&mut client, &mut server);
+
+        let headers1xx = vec![(String::from(":status"), String::from("101"))];
+
+        let encoded_headers = server
+            .encoder
+            .encode_header_block(&mut server.conn, &headers1xx, request_stream_id)
+            .unwrap();
+        let hframe = HFrame::Headers {
+            header_block: encoded_headers.to_vec(),
+        };
+        let mut d = Encoder::default();
+        hframe.encode(&mut d);
+        let headers200 = vec![
+            (String::from(":status"), String::from("200")),
+            (String::from("my-header"), String::from("my-header")),
+            (String::from("content-length"), String::from("3")),
+        ];
+
+        let encoded_headers = server
```

Comment:
```suggestion
        let encoded_200 = server
```


... the alternative is to define a function that takes a slice of header tuples and a mutable reference to the encoder and encodes those headers into the encoder.  That might be easier to read.